1_preprocessing/1_get_mission_altitude.py

```diff
@@ -0,0 +1,120 @@
+
+    valid_ratio = num_valid / num_total
+
+    print(f"Valid elevation points: {num_valid}/{num_total} ({valid_ratio:.1%})")
+
+    if valid_ratio < 0.9:
+        raise ValueError("Failed. More than 10% of camera points fall outside the DTM extent.")
+
+
+    # Step 5: Process elevations and calculate height above ground
+    heights_above_ground = []
+    ground_elevations = []
+    camera_elevations = []
+
+    for cam_pt, elev in zip(DTM_crs_cam_locations.geometry, elevations):
+        # Skip no-data points
+        if elev.mask[0] == True:
+            continue
+        # Index 0 because each sampled elevation is a 1 element masked array
+        ground_height = elev.data[0]
+        cam_height = cam_pt.z
+        # Record the difference in heights as the altitude
+        heights_above_ground.append(cam_height - ground_height)
+        ground_elevations.append(ground_height)
+        camera_elevations.append(cam_height)
+
+    # Step 6: Compute summary statistics
+    heights_np = np.array(heights_above_ground)
+    ground_np = np.array(ground_elevations)
+    camera_np = np.array(camera_elevations)
+
+    cv = np.std(heights_np) / np.mean(heights_np)
+    correlation = np.corrcoef(ground_np, camera_np)[0, 1]  # Get value from the correlation matrix
+
+    if args.verbose:
+        stats = {
+            "count": len(heights_np),
+            "mean": np.mean(heights_np),
+            "std": np.std(heights_np),
+            "min": np.min(heights_np),
+            "max": np.max(heights_np),
+            "median": np.median(heights_np),
+            "cv": cv,
+            "corr_ground_vs_camera_elevation": correlation
+        }
+
+        print("Height above ground summary stats:")
+        for k, v in stats.items():
+            print(f"{k}: {v:.2f}")
+
+    # Prepare the data as a single-row dictionary
+    summary_row = {
+        "mean_height_above_ground": np.mean(heights_np),
+        "cv_height_above_ground": cv,
+        "corr_ground_vs_camera_elevation": correlation
+    }
+
+    # Convert to DataFrame and export as CSV
+    summary_df = pd.DataFrame([summary_row])
+
+    # Ensure output directory exists and save the summary
+    ensure_containing_folder(output_csv)
+    summary_df.to_csv(output_csv, index=False)
+
+    print(f"Summary exported to {output_csv}")
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Compute height above ground for cameras and export summary statistics.")
+    parser.add_argument("--camera-file", default=EXAMPLE_CAMERAS_FILENAME, help="Path to camera file")
+    parser.add_argument("--dtm-file", default=EXAMPLE_DTM_FILE, help="Path to DTM raster file")
+    parser.add_argument("--output-csv", default="altitude_summary.csv", help="Path to save output CSV summary")
+    parser.add_argument("--verbose", action="store_true", help="Print summary statistics to stdout")
+
+    args = parser.parse_args()
+    main(args.camera_file, args.dtm_file, args.output_csv)
```

Comment:
Completely optional: one pattern I've used if I have a lot of arguments is to do `main(**args.__dict__)`. Then it's ensured that the argparse options are provided in the correct order, even if you reorder the arguments in main.